plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/githubPullRequest.ts

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import fs from 'fs';
+import path from 'path';
+import { promisify } from 'util';
+
+import {
+  GithubCredentialsProvider,
+  ScmIntegrationRegistry,
+} from '@backstage/integration';
+import { zipObject } from 'lodash';
+import { createTemplateAction } from '../../createTemplateAction';
+import { Octokit } from '@octokit/rest';
+import { OctokitResponse } from '@octokit/types';
+import { InputError } from '@backstage/errors';
+import { createPullRequest } from 'octokit-plugin-create-pull-request';
```

Comment:
This is insane how easy this actually is with github..

Reply:
It's a bit of a faff with the API to be honest - you end up juggling references to trees and other lower-level git constructs. You can see the implementation of this library here - https://github.com/gr2m/octokit-plugin-create-pull-request/tree/main/src - it's not AlphaGo but it is 300 lines of opaque mucking about.

Reply:
yeah - sorry, I meant how easy that library makes it. :)